src/app/component-library/components/checkbox/checkbox.component.html

```diff
@@ -6,6 +6,11 @@
   [class.checkbox-container-primary]="variant === 'primary'"
   [class.checkbox-container-secondary]="variant === 'secondary'"
   (click)="toggleCheck()"
+  (keydown)="handleKeyDown($event)"
+  role="checkbox"
+  [attr.tabindex]="disabled ? -1 : 0"
+  [attr.aria-checked]="isChecked"
+  [attr.aria-disabled]="disabled"
```

Comment:
If we use the `checkbox` role attribute here, browsers will mark the child elements as presentational (see: [ARIA checkbox role descendants](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/checkbox_role#all_descendants_are_presentational)). Since the label is a child element, we should really only apply this role on the actual specific element that's representing the checkbox.